openvbi/adaptors/dcdb.py

```diff
@@ -31,68 +31,81 @@
+        data = data.astype({'z':'float'})
+        data = data.dropna(subset=['t'])
 
-    meta = md.Metadata()
-    meta.setProviderID(provider, 'UNKNOWN')
-    meta.setIdentifiers(logger_uuid, 'UNKNOWN', 'UNKNOWN')
-    meta.setReferencing(md.VerticalReference.UNKNOWN, md.VerticalReferencePosition.GNSS)
-    meta.setVessel('UNKNOWN', ship_name, -1.0)
+        meta = md.Metadata()
+        meta.setProviderID(provider, 'UNKNOWN')
+        meta.setIdentifiers(logger_uuid, 'UNKNOWN', 'UNKNOWN')
+        meta.setReferencing(md.VerticalReference.UNKNOWN, md.VerticalReferencePosition.GNSS)
+        meta.setVessel('UNKNOWN', ship_name, -1.0)
 
-    return geopandas.GeoDataFrame(data, geometry=geopandas.points_from_xy(data.lon, data.lat), crs='EPSG:4326'), meta
+        return geopandas.GeoDataFrame(data, geometry=geopandas.points_from_xy(data.lon, data.lat), crs='EPSG:4326'), meta
 
-def write_geojson(dataset: Dataset, filename: str | Path, **kwargs) -> None:
-    FMT_OBS_TIME='%Y-%m-%dT%H:%M:%S.%fZ'
-    feature_lst = []
-    for n in range(len(dataset.depths)):
-        timestamp = datetime.fromtimestamp(dataset.depths['t'].iloc[n], tz=timezone.utc).strftime(FMT_OBS_TIME)
-        feature = {
-            "type": "Feature",
-            "geometry": {
-                "type": "Point",
-                "coordinates": [
-                    dataset.depths['lon'].iloc[n],
-                    dataset.depths['lat'].iloc[n]
-                ]
-            },
-            "properties": {
-                "depth": dataset.depths['z'].iloc[n],
-                "uncertainty": dataset.depths['u'].iloc[n],
-                "time": timestamp
+class GeoJSONWriter(Writer):
+    def suffix(self) -> str:
+        return '.json'
+    
+    def write(self, dataset: Dataset, filename: str | Path, **kwargs) -> None:
+        FMT_OBS_TIME='%Y-%m-%dT%H:%M:%S.%fZ'
+        feature_lst = []
+        for n in range(len(dataset.depths)):
+            timestamp = datetime.fromtimestamp(dataset.depths['t'].iloc[n], tz=timezone.utc).strftime(FMT_OBS_TIME)
+            feature = {
+                "type": "Feature",
+                "geometry": {
+                    "type": "Point",
+                    "coordinates": [
+                        dataset.depths['lon'].iloc[n],
+                        dataset.depths['lat'].iloc[n]
+                    ]
+                },
+                "properties": {
+                    "depth": dataset.depths['z'].iloc[n],
+                    "uncertainty": dataset.depths['u'].iloc[n],
+                    "time": timestamp
+                }
             }
-        }
-        feature_lst.append(dict(feature))
-    data = dataset.meta.metadata()
-    data['features'] = feature_lst
-    with open(filename, 'w') as f:
-        json.dump(data, f, **kwargs)
+            feature_lst.append(dict(feature))
+        data = dataset.meta.metadata()
+        data['features'] = feature_lst
+        with open(filename, 'w') as f:
```

Comment:
any time we are writing to JSON, we should explicitly set the encoding to utf-8 when opening the file handle (i.e. `encoding='utf-8'`). Failure to do so can cause problems on Windows whose default encoding is not utf-8.